pkg/model/awsmodel/iam.go

```diff
@@ -297,24 +288,34 @@ func (b *IAMModelBuilder) buildIAMTasks(role iam.Subject, iamName string, c *fi.
 				Name:      fi.String(iamName),
 				Lifecycle: b.Lifecycle,
 				Shared:    fi.Bool(shared),
-				Tags:      b.CloudTags(iamName, false),
+				Tags:      b.CloudTags(iamName, shared),
 			}
 			c.AddTask(iamInstanceProfile)
 		}
 
-		{
-			iamInstanceProfileRole := &awstasks.IAMInstanceProfileRole{
-				Name:      fi.String(iamName),
-				Lifecycle: b.Lifecycle,
+		// Create External Policy tasks
+		if !shared {
 
-				InstanceProfile: iamInstanceProfile,
-				Role:            iamRole,
+			iamRole, err := b.buildIAMRole(role, iamName, c)
+			if err != nil {
+				return err
 			}
-			c.AddTask(iamInstanceProfileRole)
-		}
 
-		// Create External Policy tasks
-		if !shared {
+			if err := b.buildIAMRolePolicy(role, iamName, iamRole, c); err != nil {
+				return err
+			}
+			{
+				iamInstanceProfileRole := &awstasks.IAMInstanceProfileRole{
+					Name:      fi.String(iamName),
+					Lifecycle: b.Lifecycle,
+
+					InstanceProfile: iamInstanceProfile,
+					Role:            iamRole,
+				}
+				c.AddTask(iamInstanceProfileRole)
+			}
+
+			// Create External Policy tasks
```

Comment:
Having this as a block would make it more readable, but your choice.
```suggestion
			// Create External Policy tasks
			{
```